str/runners/str_iterative_hipstr_runner_sharded.py

```diff
@@ -0,0 +1,163 @@
+
+
+def get_cloudfuse_paths(dataset, input_cpg_sids):
+    """Retrieves cloud fuse paths and outputs as a comma-separated string for crams associated with the external-wgs-ids"""
+
+    cram_retrieval_query = gql(
+        """
+        query MyQuery($dataset: String!,$input_cpg_sids: [String!]!) {
+    project(name: $dataset) {
+        sequencingGroups(id: {in_: $input_cpg_sids}) {
+        id
+        sample {
+            externalId
+        }
+        analyses(type: {eq: "cram"}, active: {eq: true}) {
+            output
+            timestampCompleted
+        }
+        }
+    }
+
+    }
+        """
+    )
+    response = query(
+        cram_retrieval_query,
+        variables={'dataset': dataset, 'input_cpg_sids': input_cpg_sids},
+    )
+    crams_by_id = {}
+    for i in response['project']['sequencingGroups']:
+        for cram in i['analyses']:
+            # ignore archived CRAMs and ONT crams
+            if cram['output'] is None:
+                continue
+            if 'archive' in cram['output']:
+                continue
+            if 'ont' in cram['output']:
+                continue
+            crams_by_id[i['id']] = cram
+
+    if len(crams_by_id) != len(input_cpg_sids):
+        cpg_sids_without_crams = set(input_cpg_sids) - set(crams_by_id.keys())
+        logging.warning(
+            f'There were some samples without CRAMs: {cpg_sids_without_crams}'
+        )
+
+    # Create string containing paths based on /cramfuse
+    cramfuse_path = []
+    for cram_obj in crams_by_id.values():
+        suffix = cram_obj['output'].removeprefix('gs://').split('/', maxsplit=1)[1]
+        cramfuse_path.append(f'/cramfuse/{suffix}')
+    cramfuse_path = ','.join(cramfuse_path)  # string format for input into hipstr
+    return cramfuse_path
+
+
+# inputs:
+@click.option(
+    '--job-storage', help='Storage of the Hail batch job eg 30G', default='30G'
+)
+@click.option('--job-memory', help='Memory of the Hail batch job', default='highmem')
+@click.option(
+    '--variant-catalog',
+    help='Full path to HipSTR Variants sharded catalog directory or file path to unsharded catalog',
+)
+@click.option('--dataset', help='dataset eg tob-wgs')
+@click.argument('internal-cpg-ids', nargs=-1)
+@click.option('--output-file-name', help='Output file name without file extension')
+@click.command()
+def main(
+    job_storage,
+    job_memory,
+    variant_catalog,
+    dataset,
+    internal_cpg_ids,
+    output_file_name,
+):  # pylint: disable=missing-function-docstring
+    b = get_batch()
+    ref_fasta = 'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
+    # Read in reference
+    ref = b.read_input_group(
+        **dict(
+            base=ref_fasta,
+            fai=ref_fasta + '.fai',
+            dict=ref_fasta.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+            + '.dict',
+        )
+    )
+
+    catalog_files = list(to_path(variant_catalog).glob('*.bed'))
```

Comment:
you might be able to run enumerate on the results of the `glob` directly, but this is fine